File: snippets/dart_snippets/lib/fiat_currencies.dart
import 'package:dart_snippets/sdk_instance.dart';
import 'package:breez_sdk/bridge_generated.dart';

Future<List<FiatCurrency>> listFiatCurrencies() async {
  // ANCHOR: list-fiat-currencies
  List<FiatCurrency> fiatCurrencyList = await breezSDK.listFiatCurrencies();
  // ANCHOR_END: list-fiat-currencies
  return fiatCurrencyList;
}

Future<Map<String, Rate>> fetchFiatRates(String lspId) async {
  // ANCHOR: fetch-fiat-rates
  Map<String, Rate> fiatRatesMap = await breezSDK.fetchFiatRates();
  // print your desired rate
  print(fiatRatesMap["USD"]?.value);
  // ANCHOR_END: fetch-fiat-rates
  return fiatRatesMap;
}

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/FiatCurrencies.kt
package com.example.kotlinmpplib

import breez_sdk.*

class FiatCurrencies {
    fun listFiatCurrencies(sdk: BlockingBreezServices) {
        // ANCHOR: list-fiat-currencies
        try {
            val fiatCurrencyList = sdk.listFiatCurrencies()
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: list-fiat-currencies
    }

    fun fetchFiatRates(sdk: BlockingBreezServices) {
        // ANCHOR: fetch-fiat-rates
        try {
            val fiatRatesMap = sdk.fetchFiatRates()
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: fetch-fiat-rates
    }
}